File: rust/src/richest_customer_wealth.rs
fn maximum_wealth(accounts: Vec<Vec<i32>>) -> i32 {
    accounts.iter().map(|x| x.iter().sum()).max().unwrap()
}

#[cfg(test)]
mod tests {
    use std::vec;

    use super::*;
    use rstest::rstest;

    #[rstest]
    #[case(vec![vec![1, 2, 3], vec![3, 2, 1]], 6)]
    #[case(vec![vec![1, 5], vec![7, 3], vec![3, 5]], 10)]
    #[case(vec![vec![2, 8, 7], vec![7, 1, 3], vec![1, 9, 5]], 17)]
    fn test_maximum_wealth(#[case] accounts: Vec<Vec<i32>>, #[case] expected: i32) {
        let result = maximum_wealth(accounts);
        assert_eq!(result, expected);
    }
}

File: cpp/src/richest_customer_wealth.cpp
#include <vector>
#include <numeric>

int maximumWealth(std::vector<std::vector<int>> &accounts)
{
	return std::transform_reduce(
		accounts.cbegin(),
		accounts.cend(),
		0,
		[](auto a, auto b)
		{ return std::max(a, b); },
		[](auto const &row)
		{ return std::reduce(row.cbegin(), row.cend()); });
}